src/protocol/boolean/solved_bits.rs

```diff
@@ -75,20 +76,21 @@ impl SolvedBits {
         ctx: ProtocolContext<'_, Replicated<F>, F>,
         record_id: RecordId,
     ) -> Result<Vec<Replicated<F>>, Error> {
-        // We assume the bit length we operate in would not exceed 255
-        #[allow(clippy::cast_possible_truncation)]
-        let l = F::Integer::BITS as u8;
+        // Calculate the number of bits we need to form a random number that
+        // has the same number of bits as the prime.
+        let l = u128::BITS - F::PRIME.into().leading_zeros();
```

Comment:
`const PRIME` of a generic `F` is of a type `Integer`, and only supports `Into<u128>`. Subtract from 128 to get the number of bits needed to generate.